### PR TITLE
add log4j implementation

### PR DESCRIPTION
**Pull request by jtmg**
# What this pull request does?  
This pull request is responsible by implementing the log4j to be used in logging. When this tool is executed as an executable the log4j lib is configured to log in the console, and when executed as a dependency is logged on file (einstein.log) that resides in the directory that is  being run.  
# How to test?  
To test as an executable you need to the tool with the verbose parameter. When passed the debug level is set to `DEBUG` and if not passed the level is `INFO` (eg. `.\einstein.exe -p middleware/server-parent:3.20.0 -v`).  
To test as a lib, you need to create a dummy project using gradle and import the `einstein lib as such, compile group: 'io.github.asseco-pst', name: 'einstein', version: '3.0.0-SNAPSHOT'` and don't forget to define the oss repository as such `maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }`

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,5 +1,5 @@
 group 'io.github.asseco-pst'
-version "1.0.0-SNAPSHOT"
+version "1.1.0-SNAPSHOT"
```
Comment (hgpestana) [inline build.gradle:2]: @jtmg validar a questão da versão

File: build.gradle [hunk 2]
```diff
@@ -14,11 +14,12 @@ buildscript {
 
 apply plugin: 'groovy'
 apply plugin: 'signing'
+apply plugin: 'maven-publish'
 apply plugin: 'java'
 apply plugin: 'application'
 apply plugin: 'com.github.johnrengelman.shadow'
 apply plugin: 'edu.sc.seis.launch4j'
-apply plugin: 'com.bmuschko.nexus'
+//apply plugin: 'com.bmuschko.nexus'
```
Comment (hgpestana) [inline build.gradle:22]: @jtmg vale a pena manter comentado?

File: build.gradle [hunk 3]
```diff
@@ -58,41 +61,68 @@ createExe {
...
+publishing {
+    publications {
+        shadow(MavenPublication) {
+            artifactId = 'einstein'
+            project.shadow.component(it)
+            pom {
+                name = 'einstein'
+                description = 'A library that allows managing artifacts in a Liberty server'
+                url = 'https://github.com/asseco-pst/einstein'
+                inceptionYear = '2020'
```
Comment (hgpestana) [inline build.gradle:87]: @jtmg convêm validar

**Review by hgpestana — COMMENTED** [review]
Bom dia @jtmg.

Parece-me tudo ok, só deixei alguns comentários nalgumas situações que são necessárias rever.

Obrigado

**Comment by hgpestana** [comment]
@jtmg fiz os testes necessários e o comportamento foi o esperado. Deixo então só as notas acima mencionadas.

Obrigado

File: build.gradle (hunk 2 above)
Comment (jtmg) [reply build.gradle:22]: Esse plugin era utilizado para fazer deploy dos artefacto para o repositório. agora como normalizamos o deploy para ficar igual ao vulture este plugin tem de ser removido

**Pull request merged**